### ISSUE TITLE
OmniSharp analysis is extremely slow on only one small-ish project out of four

### ISSUE THREAD

**Issue by alexrp** [OmniSharp/omnisharp-roslyn#2295]
## Versions

* VS Code: 1.62.3
* C# Extension: 1.23.17
* OmniSharp: 1.37.18-beta.83

## Repro

1. `git clone git@github.com:alexrp/system-terminal.git`
2. `cd system-terminal && git checkout d28f485e05d1557abc690a4f0316ef92e5f023c1 && dotnet build`
3. Open the repository root as a folder in VS Code
4. Open `src/core/Terminal.cs`
5. Start making arbitrary edits

At this point you should notice that OmniSharp's analysis (notably, *not* completion) is extremely slow to respond to edits. Style diagnostics (both from the `IDExxxx` analyzers and StyleCop) as well as code quality diagnostics (`CAxxxx` analyzers) can sometimes seem to get completely stuck to the point where you have to close the file and reopen it for analysis to start over and finally respond to the changes you've made.

What is interesting is that the same is *not* true in other projects in the repository. For example, if you open `src/extensions/Hosting/TerminalHost.cs` and start making edits, OmniSharp immediately goes back to normal with analysis being virtually instantaneous.

## Log

Worth noting that the OmniSharp log is full of stuff like this when editing the problematic project:

```
Received response for /completion but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /completion/resolve but could not find request.
[warn]: OmniSharp.Stdio.Host
        ************ Request ************
{
  "Type": "request",
  "Seq": 186,
  "Command": "/filesChanged",
  "Arguments": [
    {
      "FileName": "c:\\Users\\alex\\source\\repos\\system-terminal\\src\\core\\Terminal.cs",
      "changeType": "Change"
    }
  ]
}
[fail]: OmniSharp.Stdio.Host
        ************  Response (9.6956ms) ************ 
{
  "Request_seq": 186,
  "Command": "/filesChanged",
  "Running": true,
  "Success": false,
  "Message": "\"System.IO.IOException: The process cannot access the file 'c:\\\\Users\\\\alex\\\\source\\\\repos\\\\system-terminal\\\\src\\\\core\\\\Terminal.cs' because it is being used by another process.\\r\\n   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)\\r\\n   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)\\r\\n   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)\\r\\n   at System.IO.StreamReader..ctor(String path, Encoding encoding, Boolean detectEncodingFromByteOrderMarks, Int32 bufferSize, Boolean checkHost)\\r\\n   at System.IO.File.InternalReadAllText(String path, Encoding encoding, Boolean checkHost)\\r\\n   at OmniSharp.Roslyn.BufferManager.<UpdateBufferAsync>d__8.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Roslyn\\\\BufferManager.cs:line 47\"",
  "Body": null,
  "Seq": 229,
  "Type": "response"
}
Received response for /v2/getcodeactions but could not find request.
Received response for /quickinfo but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
Received response for /v2/getcodeactions but could not find request.
[warn]: OmniSharp.Roslyn.CSharp.Services.Navigation.FindUsagesService
        No symbol found. File: c:\Users\alex\source\repos\system-terminal\src\core\Terminal.cs, Line: 57, Column: 48.
[warn]: OmniSharp.Roslyn.CSharp.Services.Navigation.FindUsagesService
        No symbol found. File: c:\Users\alex\source\repos\system-terminal\src\core\Terminal.cs, Line: 0, Column: 9.
Received response for /v2/getcodeactions but could not find request.
[warn]: OmniSharp.Stdio.Host
        ************ Request ************
{
  "Type": "request",
  "Seq": 270,
  "Command": "/v2/getcodeactions",
  "Arguments": {
    "FileName": "c:\\Users\\alex\\source\\repos\\system-terminal\\src\\core\\Terminal.cs",
    "Line": 57,
    "Column": 43
  }
}
[fail]: OmniSharp.Stdio.Host
        ************  Response (5343.5150ms) ************ 
{
  "Request_seq": 270,
  "Command": "/v2/getcodeactions",
  "Running": true,
  "Success": false,
  "Message": "\"System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.\\r\\n   at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)\\r\\n   at System.Reflection.RuntimeAssembly.get_DefinedTypes()\\r\\n   at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext()\\r\\n   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()\\r\\n   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()\\r\\n   at System.Linq.Enumerable.<ConcatIterator>d__59`1.MoveNext()\\r\\n   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)\\r\\n   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)\\r\\n   at System.Collections.Immutable.ImmutableArray.CreateRange[T](IEnumerable`1 items)\\r\\n   at OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.CachingCodeFixProviderForProjects.LoadFrom(Project project) in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Roslyn.CSharp\\\\Services\\\\Refactoring\\\\V2\\\\CachingCodeFixProviderForProjects.cs:line 93\\r\\n   at OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.CachingCodeFixProviderForProjects.GetAllCodeFixesForProject(ProjectId projectId) in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Roslyn.CSharp\\\\Services\\\\Refactoring\\\\V2\\\\CachingCodeFixProviderForProjects.cs:line 57\\r\\n   at OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.BaseCodeActionService`2.GetSortedCodeFixProviders(Document document) in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Roslyn.CSharp\\\\Services\\\\Refactoring\\\\V2\\\\BaseCodeActionService.cs:line 156\\r\\n   at OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.BaseCodeActionService`2.<AppendFixesAsync>d__13.MoveNext()\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.BaseCodeActionService`2.<CollectCodeFixesActions>d__12.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Roslyn.CSharp\\\\Services\\\\Refactoring\\\\V2\\\\BaseCodeActionService.cs:line 123\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.BaseCodeActionService`2.<GetAvailableCodeActions>d__9.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Roslyn.CSharp\\\\Services\\\\Refactoring\\\\V2\\\\BaseCodeActionService.cs:line 80\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.GetCodeActionsService.<Handle>d__1.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Roslyn.CSharp\\\\Services\\\\Refactoring\\\\V2\\\\GetCodeActionsService.cs:line 32\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Endpoint.EndpointHandler`2.<GetFirstNotEmptyResponseFromHandlers>d__19.MoveNext()\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Endpoint.EndpointHandler`2.<HandleRequestForLanguage>d__20.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Host\\\\Endpoint\\\\EndpointHandler.cs:line 230\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Endpoint.EndpointHandler`2.<Process>d__16.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Host\\\\Endpoint\\\\EndpointHandler.cs:line 131\\r\\n--- End of stack trace from previous location where exception was thrown ---\\r\\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\\r\\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\\r\\n   at OmniSharp.Stdio.Host.<HandleRequest>d__14.MoveNext() in D:\\\\a\\\\1\\\\s\\\\src\\\\OmniSharp.Stdio\\\\Host.cs:line 218\"",
  "Body": null,
  "Seq": 322,
  "Type": "response"
}
Received response for /v2/getcodeactions but could not find request.
```

Not sure what's up with the `IOException`; I'm certainly not editing the file in other programs. Also, I'm not sure how to debug that `ReflectionTypeLoadException` further.

**Comment by alexrp** [comment]
FWIW, I believe this started happening a little over a week ago, but it's hard to tell if it started happening due to an OmniSharp change or a change in my project...

**Comment by filipw** [comment]
Does this happen without the beta OmniSharp - on the current released `1.37.17` that is by default in the extension?
I cloned your project and there are indeed some slow code actions and code check calls.

Did you install Stylecop Analyzers into the project recently? Looks to me like most of the slowness is coming from there.

**Comment by alexrp** [comment]
> Does this happen without the beta OmniSharp - on the current released `1.37.17` that is by default in the extension?

Nope. Analysis seems to work fine with the stable version.

> Did you install Stylecop Analyzers into the project recently? Looks to me like most of the slowness is coming from there.

No, StyleCop has been there all along. Removing both StyleCop and CsWin32 from the project makes no difference.

**Comment by filipw** [comment]
Thanks.

@JoeRobich since this happens on the pre-release only, and not on latest stable, the only possible change that could cause a regression is https://github.com/OmniSharp/omnisharp-roslyn/pull/2267  Would you have some time to check the above repro project against this change?

**Comment by filipw** [comment]
This has been resolved by reverting #2267
